packages/core-data/src/selectors.ts

```diff
@@ -1129,7 +1131,10 @@ export function getAutosave< EntityRecord extends ET.EntityRecord< any > >(
 	}
 
 	const autosaves = state.autosaves[ postId ];
-	return find( autosaves, { author: authorId } ) as EntityRecord | undefined;
+
+	return autosaves?.find(
+		( autosave: any ) => autosave.author === authorId
```

Comment:
We're using `any` here to match the pre-existing type of data that the `getAutosaves()` selector is returning. This, of course, needs tweaking but is a separate task.